enhancements/microshift/gateway-api-support.md

```diff
@@ -0,0 +1,434 @@
+released as dev preview, users need to opt into it. Additional memory should be
+planned when enabling this feature, as OSSM components and the gateways
+themselves consume additional resources.
+
+## Open Questions [optional]
+N/A
+
+## Test Plan
+Gateway API support for MicroShift will be tested using existing test
+harness. A new suite will be created with a simple test for the overall setup
+with an HTTP route. Another test will ensure compatibility with the current
+OpenShift router deployment.
+
+## Graduation Criteria
+Gateway API for MicroShift is targeted to be Dev Preview on MicroShift 4.18.
+
+At the time of this writing OSSM is in Tech Preview state, expected to graduate
+to GA in 2025.
+
+Graduation criteria will be determined by customer adoption/feedback.
+
+### Dev Preview -> Tech Preview
+N/A
+
+### Tech Preview -> GA
+N/A
+
+### Removing a deprecated feature
+N/A
+
+## Upgrade / Downgrade Strategy
+Gateway API and MicroShift RPMs get built from the same spec so they share
+version. Whenever an upgrade happens both of them are also upgraded at the
+same time.
+
+The bulk of manifests are CRDs which have their own versioning, so the
+upgrade/downgrade strategy relies on the components using them. Extracting the
+OSSM manifests from the bundle guarantees that Istio will be compatible with
+them.
+
+For Gateway and HTTPRoute/GRPCRoute resources, manual action may be needed but
+this is the application's responsibility.
+
+## Version Skew Strategy
+Building Gateway API and MicroShift RPMs from the same spec file means both
+should be updated together which means there should not be any version skew
+between them.
+
+## Operational Aspects of API Extensions
+N/A
+
+## Support Procedures
+In order to debug error situations with Gateway API and OSSM, Gateways and
+routes need to be inspected.
+
+The first thing to check is the gatewayclass status, it needs to be accepted
+and handled by the istio controller, as shown in the status:
+```
+$ oc get gatewayclass openshift-gateway-api -o yaml
+apiVersion: gateway.networking.k8s.io/v1
+kind: GatewayClass
+metadata:
+  creationTimestamp: "2024-10-14T07:44:20Z"
+  generation: 1
+  name: openshift-gateway-api
+  resourceVersion: "1060"
+  uid: 0707b37e-c3c6-4b8a-a344-24545d8b8351
+spec:
+  controllerName: openshift.io/gateway-controller
+status:
+  conditions:
+  - lastTransitionTime: "2024-10-14T07:44:30Z"
+    message: Handled by Istio controller
+    observedGeneration: 1
+    reason: Accepted
+    status: "True"
+    type: Accepted
+```
+
+If we have a Gateway `demo-gateway` using the above GatewayClass we should see it has been accepted and programmed by the gateway class:
+```
+$ oc get gateway -n openshift-ingress demo-gateway -o yaml
+apiVersion: gateway.networking.k8s.io/v1
+kind: Gateway
+metadata:
+  annotations:
+    gateway.istio.io/controller-version: "5"
+  creationTimestamp: "2024-10-14T07:48:16Z"
+  generation: 1
+  name: demo-gateway
+  namespace: openshift-ingress
```

Comment:
The `namespace` you use will be very important in terms of access between Gateways/Routes/Services, but should not be `openshift-ingress`.

Reply:
Updated to `openshift-gateway-api`